.github/CONTRIBUTING.md

```diff
@@ -175,64 +213,56 @@ There are checks that automation will not be able to capture. For example:
 
-## Fixing bug
+### Fixing bugs
 
 Write the bug repro as a test, before fixing the bug.
 
-1. Clone/prepare the repository or reset an existing repository
-   -  To reset the repo, run `npm run tableflip`
-1. Run continuous build, run `npm start`
-1. Convert bug into a test case, under `__tests__/html/this-is-the-bug.html`
-1. Run the test, `npm run start:docker`, then `npm test -- --testPathPattern html/this-is-the-bug.html`
+1. Clone and prepare the repository or reset an existing one
+   -  To reset, run `npm run tableflip`
+1. Run continuous build by running `npm start`
+1. Reproduce the bug on a test page, under `__tests__/html/this-is-the-bug.html`
+1. Run `npm run browser` and navigate to the test page
    -  Make sure the test fail as described in the repro
 1. Fix the bug
-1. Re-run `npm run start:docker` to refresh the code inside Docker
-1. Re-run the test, it should succeed
-   -  Verify screenshots if any
-1. Add a log entry to `CHANGELOG.md`
-1. File a pull request
+1. Reload the test page, it should succeed
+1. Run all test suites in an automated manner
+   -  Run `npm run docker`, followed by `npm test` in a new terminal
+1. Do [final checks](#final-checks)
+1. Submit a pull request
 
-## Implementing new features
+### Implementing new features
 
 Write the user story while implementing the feature.
 
-1. Clone/prepare the repository or reset an existing repository
-   -  To reset the repo, run `npm run tableflip`
-1. Run continuous build, run `npm start`
-1. Clone `__tests__/html/simple.html` to `my-feature.html` and use it as a playground
-1. Navigate to the new playground at http://localhost:5001/my-feature.html
-1. Implement the feature while modifying the playground
-1. After the feature is completed, write more tests to include cases that need more attention, and unhappy paths
-1. Run the test, `npm run start:docker`, then `npm test -- --testPathPattern html/my-feature*.html`
-   -  Verify screenshots
-1. Write a new sample with `README.md`
-   -  This is the user story and proof-of-record on how the feature will work
-   -  Update [`samples/README.md`](https://github.com/microsoft/BotFramework-WebChat/blob/main/samples/README.md)
-1. Add more design docs to [`/docs`](https://github.com/microsoft/BotFramework-WebChat/tree/main/docs) as needed
-1. Add a few log entries to `CHANGELOG.md`
-1. File a pull request
+1. Clone and prepare the repository or reset an existing one
+   -  To reset, run `npm run tableflip`
+1. Run continuous build by running `npm start`
+1. Clone the test page `__tests__/html/simple.html` to `my-feature.html` and use it as a playground
+1. Run `npm run browser` and navigate to the test page
+1. Implement the feature and update/add test pages as needed
+   -  Test pages should include cases that need more attention, and unhappy paths
+1. Run all test suites in an automated manner
+   -  Run `npm run docker`, followed by `npm test` in a new terminal
```

Comment:
See notes above for corrections

Reply:
Please check again if I have applied all of them.